src/mds/Server.cc

```diff
@@ -6450,6 +6450,11 @@ int Server::xattr_validate(CInode *cur, const InodeStoreBase::xattr_map_const_pt
       return -CEPHFS_ENODATA;
     }
 
+    if ((flags & CEPH_XATTR_REMOVE2) && !(xattrs && xattrs->count(mempool::mds_co::string(xattr_name)))) {
```

Comment:
Is this new flag _really_ required? Can't we specially check for POSIX ACLs and signal success? For other xattrs, the MDS will do the right thing and return -ENODATA.

Reply:
@vshankar Do you mean do this special check in client side, right ?

Currently we need to make sure with this PR it still could work with the old clients, which haven't support the kclient patch or the `Client.cc` fixing yet. The old clients will continue send the `CEPH_XATTR_REMOVE` flags but they won't do the ACLs check, and then they will fail as we hit in https://tracker.ceph.com/issues/64679#note-4.

Reply:
> @vshankar Do you mean do this special check in client side, right ?
> 
> Currently we need to make sure with this PR it still could work with the old clients, which haven't support the kclient patch or the `Client.cc` fixing yet.

Sure. Old clients assume removing posix acl xattrs even if those don't exist would return 0.

> The old clients will continue send the `CEPH_XATTR_REMOVE` flags but they won't do the ACLs check, and then they will fail as we hit in https://tracker.ceph.com/issues/64679#note-4.

Sure. If the MDS can return 0 for posix acls even if those don't exist, isn't that behaviour preserved with oder client?

Maybe I'm missing something @lxbsz

Reply:
I do think for the ACLs being specially handled should be more like a client side manner. I didn't find any POSIX is saying what we should do. But I am okay if you prefer to do this in MDS. Please let me know and then I will revise this.

Reply:
I see what you mean -- the MDS does not have special handling for ACL_EA_ACCESS/ACL_EA_DEFAULT. So, we are left with doing it this way...